docs/developer-guide/writing-web-integration-tests.md

```diff
@@ -0,0 +1,142 @@
+* Below are the steps which should be followed for tests:
+  * Create a groovy based [test class](https://github.com/strongbox/strongbox-web-integration-tests/blob/master/pypi/src/it/common-flows/test-pypi-common-flows.groovy) for the test cases.
+  * Import Base test class.
+    ```java
+    def baseScript = new GroovyScriptEngine("$project.basedir/src/it").with
+    { 
+        loadScriptByName('PypiWebIntegrationTest.groovy') 
+    }
+    ```
+  * Determine Operation system and decide command to be run based on OS.
+  * Execute the command and verify/assert the result.
+    ```java
+    // Assert directory exists for package uploaded
+    Path packageDirectoryPath = uploadPackageDirectoryPath.resolve("dist");
+    boolean pathExists = Files.isDirectory(packageDirectoryPath,
+                                           LinkOption.NOFOLLOW_LINKS);
+    
+    assert pathExists == true
+              
+    // upload python package using pip command
+    runCommand(uploadPackageDirectoryPath, packageUploadCommand)
+              
+    def commandOutput
+    // Install uploaded python package using pip command and assert success
+    def uploadedPackageName = "pip_upload_test"
+    commandOutput = runCommand(uploadPackageDirectoryPath, pipInstallPackageCommand + " " + uploadedPackageName)
+    assert commandOutput.contains("Successfully installed " + uploadedPackageName.replace("_" , "-") + "-1.0")
+    ``` 
+      
+## Running Tests
+
+The Web Integration tests can be run using both Maven and Docker.
+
+Typically, you would use Maven to run the tests from one sub-module, (unless you have all the possible tools that are required for executing the tests, for example, Gradle, NuGet, NPM already installed, which is usually not the case). This is why re recommend running the tests using Docker, as the Docker images are pre-configured with everything you will need to run the tests in the exact same way they would run on our CI server (Jenkins).
+
+### Using Maven
+
+Each module has a `pom.xml` file. Go to the respective module (for example. `cd pypi`) and execute the `mvn clean install` command. This will launch the Strongbox application instance and will execute web integration tests against it.
+
+### Using Docker
+
+To make development and testing easier, we've created Docker images for all of the tools we are testing against, so that 
+you don't have to go through the process of installing them manually.
+
+For this to work, you need to have installed `docker` and `docker-compose` on your machine.
+
+Afterwards, just go to the respective sub-project (i.e. `cd maven`) and execute `docker-compose up`.
+
+Each module has a `docker-compose.yml` file.
+
+You can run all the tests (for all modules) like this:
+
+```
+for tool in `ls -ap | grep \/ | grep -v -e "\."`; do cd ${tool} && docker-compose up & cd -; done
```

Comment:
This command will result in multiple containers outputting logs into the console at the same time, which makes it impossible to follow what's actually happening.  We can improve the original command with this one:  

```suggestion
find . -maxdepth 2 -type f -name "docker-compose.y*ml" -exec docker-compose -f {} up \;
```

Reply:
Fixed

Reply:
Thanks! :)